### PR TITLE
Fix for new hoomd dlext

### PR DESCRIPTION
**Pull request by InnocentBug**
Adjustments to PySAGES that are required with the hoomd-dlext backend.

### CODE REVIEW

File: pysages/backends/hoomd.py [hunk 1]
```diff
@@ -41,7 +40,7 @@
 CONTEXTS_SAMPLERS = {}
 
 
-class Sampler(DLextSampler):
+class Sampler(DLExtSampler):
     def __init__(self, sysdef, method_bundle, bias, dt, callback: Callable):
```
Comment (pabloferz) [inline pysages/backends/hoomd.py:44]:
```suggestion
    def __init__(self, sysview, method_bundle, bias, dt, callback: Callable):
```

**Review by pabloferz — COMMENTED** [review]
I the constructor takes a `SystemView`

File: pysages/backends/hoomd.py [hunk 2]
```diff
@@ -202,21 +199,20 @@ def bind(
...
-    context.integrator.cpp_integrator.setHalfStepHook(sampler)
-    #
-    CONTEXTS_SAMPLERS[context] = sampler
-    #
+        snapshot = take_snapshot(wrapped_context)
+        method_bundle = sampling_method.build(snapshot, helpers)
+        sync_and_bias = partial(bias, sync_backend=sysview.synchronize)
+
+        sampler = Sampler(
+            context.system_definition, method_bundle, sync_and_bias, context.integrator.dt, callback
```
[the remark this answers is not in the document]
Comment (InnocentBug) [reply pysages/backends/hoomd.py]: solved differently.

File: pysages/backends/hoomd.py (hunk 1 above)
Comment (InnocentBug) [reply pysages/backends/hoomd.py]: solved

File: pysages/backends/hoomd.py [hunk 3]
```diff
@@ -71,7 +70,7 @@ def python_update(positions, vel_mass, rtags, imgs, forces):
             if self.callback:
                 self.callback(snap, self.state, 0)
 
-        super().__init__(sysdef, python_update)
+        super().__init__(sysview, python_update, AccessMode.Read, default_location())
```
Comment (pabloferz) [inline pysages/backends/hoomd.py:73]:
```suggestion
        super().__init__(sysview, python_update, default_location(), AccessMode.Read)
```

**Review by pabloferz — CHANGES_REQUESTED** [review]
There's still a spelling commit hanging around.

**Comment by InnocentBug** [comment]
I reversed the spelling thing

**Review by pabloferz — APPROVED** [review]

**Pull request merged**